holonote/annotate/annotator.py

```diff
@@ -343,8 +343,50 @@ def get_display(self, *kdims: str | hv.Dimension) -> AnnotationDisplay:
             self._displays[element_key] = self._create_annotation_element(element_key)
         return self._displays[element_key]
 
-    def __mul__(self, other: hv.Element) -> hv.Overlay:
-        return other * self.get_element(*other.kdims)
+    def __mul__(self, other: hv.Element | hv.Layout | hv.Overlay | hv.NdOverlay) -> hv.Overlay:
+        if isinstance(other, (hv.Overlay, hv.NdOverlay)):
+            kdims, opts = other.kdims, other.opts.get().kwargs
+            if not kdims or kdims == ["Element"]:  # overlay and ndoverlay with no added kdims
+                # If no kdims in the overlay we use the first available
+                kdims = next(k for el in other.values() if (k := el.kdims))
+            return (
+                other * self.get_element(*[kdim for kdim in kdims if kdim.name in self.spec])
+            ).opts(**opts)
+        elif isinstance(other, hv.Layout):
+            opts = other.opts.get().kwargs
+            to_layout = []
+            for el in other:
+                kdims = el.kdims
+                el_opts = el.opts.get().kwargs
+                if not el.kdims or el.kdims == ["Element"]:
+                    try:
+                        kdims = next(k for sub_el in el.values() if (k := sub_el.kdims))
+                    except (
+                        StopIteration
+                    ):  # if the dmap hasn't been rendered yet, render to access kdims
+                        hv.render(el)
+                        kdims = next(k for sub_el in el.values() if (k := sub_el.kdims))
+                to_layout.append(
+                    (
+                        el * self.get_element(*[kdim for kdim in kdims if kdim.name in self.spec])
+                    ).opts(**el_opts)
```

Comment:
ok, I pushed some changes but I'm stuck and could use your expertise to finish this off...

I need to reapply the element's opts (`opts(**el_opts)`) onto the overlay in order to have opts propagate. Otherwise, I get the following, where the top plot had opts like `responsive=True` and `show_legend=False`, which did not make it when combined with the annotator. 

<img width="1268" alt="image" src="https://github.com/holoviz/holonote/assets/6613202/b0d0c9eb-e651-4bf9-93ce-00be36ff442e">

However, this breaks one of tests, I think because not all opts are valid for an overlay. Do you know of a way to apply only valid opts or efficiently filter for the valid set of opts given an element type? I'm having trouble making a minimal version of this as a test and could use guidance.